src/XMonad/Core.hs

```diff
@@ -670,6 +671,22 @@ compile dirs method =
         CompileStackGhc stackYaml -> ("stack", "ghc" : "--stack-yaml" : stackYaml : "--" : ghcArgs)
         CompileScript script -> (script, [binFileName dirs])
 
+-- | Check GHC output for deprecation warnings and notify the user if there
+-- were any. Report success otherwise.
+checkCompileWarnings :: Directories -> IO ()
+checkCompileWarnings dirs = do
+    ghcErr <- readFile (errFileName dirs)
+    if "-Wdeprecations" `isInfixOf` ghcErr
+      then do
+        let msg = unlines $
+                ["Deprecations detected while compiling xmonad config: " <> srcFileName dirs]
+                ++ lines ghcErr
+                ++ ["","Please correct them or silence using {-# OPTIONS_GHC -Wno-deprecations #-}."]
+        trace msg
+        xmessage msg
```

Comment:
I wonder if we should do something a little bit more clever here and extract the actual deprecation warnings in order to not overwhelm the user with the whole compilation output

Reply:
It's invoked with `-v0` and without `-Wall`, so there isn't a whole lot of “compilation output” to speak about. :-)

Reply:
Ah good point; especially since we're only compiling the file itself